### PR TITLE
LibGfx/ILBM: Fix multiple fuzzer found issues

### PR DESCRIPTION
**Pull request by tcl3**
This fixes oss fuzz issues: [63352](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=63352), [63355](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=63355), [63426](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=63426), [63509](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=63509) and [63711](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=63711) in addition to several issues found via local fuzzing.

### CODE REVIEW

File: Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp [hunk 1]
```diff
@@ -307,6 +337,16 @@ static ErrorOr<void> decode_bmhd_chunk(ILBMLoadingContext& context)
         return Error::from_string_literal("IFFImageDecoderPlugin: Not enough data for header chunk");
 
     context.bm_header = *bit_cast<BMHDHeader const*>(first_chunk.data.data());
+
+    if (context.bm_header.planes > 8)
```
Comment (tcl3) [inline Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp:341]: I couldn't find any documentation which places a hard limit (below 255) on the number of bitplanes, but we do calculations later on that fail if the number of bitplanes is > 8. I also couldn't find any examples of files that use more than 8 bitplanes.

[This document](https://web.archive.org/web/20140613000343/http://home.comcast.net/~erniew/lwsdk/docs/filefmts/ilbm.html) suggests that bitplanes may be ignored if they are not required, so we could possibly do that instead.

Comment (warpdesign) [reply Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp:341]: There can be files with 24 bitplanes actually. They are called "deep" and contain 24 bitplanes with full RGB data.

There can be a mask bitplane as well but I am not sure it is supported for deep files.

Comment (warpdesign) [reply Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp:341]: I have opened a [PR](https://github.com/SerenityOS/serenity/pull/21324) to add HAM support.

24bit support was my next task.

Comment (tcl3) [reply Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp:342]: I think this check is probably worth keeping for now. I've updated the error message to be a bit more meaningful. We can update/remove the check when support for deep ILBMs is added.

**Pull request merged**